base/ryu/Ryu.jl

```diff
@@ -90,10 +87,13 @@ function Base.show(io::IO, x::T) where {T <: Base.IEEEFloat}
         x = round(x, sigdigits=6)
     end
     buf = Base.StringVector(neededdigits(T))
-    pos = writeshortest(buf, 1, x)
-    @assert pos - 1 <= length(buf)
+    typed = !get(io, :compact, false) && get(io, :typeinfo, Any) != typeof(x)
+    pos = writeshortest(buf, 1, x, false, false, true, -1,
+        x isa Float32 ? UInt8('f') : UInt8('e'), false, UInt8('.'), typed)
     write(io, resize!(buf, pos - 1))
     return
 end
 
+Base.print(io::IO, x::Union{Float16, Float32}) = show(IOContext(io, :compact => true), x)
```

Comment:
Does this cause any slowdown to `string(x)` or `print(x)`? Would be a shame to give back any of the great speedup from the new code :)